src/test/java/com/hillel/items_exchange/util/PatternHandlerTest.java

```diff
@@ -0,0 +1,304 @@
+
+        return correctEmails;
+    }
+
+    private static List<String> createWrongEmails() {
+        List<String> wrongEmails = new ArrayList<>();
+        wrongEmails.add("pushkin@ukr.n");
+        wrongEmails.add(".145@gmail.com");
+        wrongEmails.add("pushkin@@mail.ru");
+        wrongEmails.add("1@111.11");
+        wrongEmails.add("pushkin.11");
+        wrongEmails.add("pushkin.1u");
+        wrongEmails.add("pushkin.u1");
+        wrongEmails.add("1@111.1u");
+        wrongEmails.add("1@111.1ru");
+        wrongEmails.add("1@111.ru1");
+        wrongEmails.add("1@ma~il.ru");
+
+        List<Character> correctCharacters = List.of('.', '_', '+', '-');
+
+        wrongEmails.addAll(getStringsWithSpecialSymbol("pu", "shkin@ukr.net", correctCharacters));
+        wrongEmails.addAll(getStringsWithRussianOrUkrCharacter("pu", "shkin@ukr.net"));
+
+        return wrongEmails;
+    }
+
+    private static List<String> createCorrectPasswords() {
+        List<String> correctPasswords = new ArrayList<>();
+        correctPasswords.add("aA1");
+        correctPasswords.add("1aA");
+        correctPasswords.add("a1A");
+        correctPasswords.add("#aA1");
+        correctPasswords.add("aA1@");
+        correctPasswords.add("~aA1`");
+        correctPasswords.add("~`aA1`!@#$%^&*()_+-=");
+
+        correctPasswords.addAll(getStringsWithSpecialSymbol("a", "A1", Collections.emptyList()));
+
+        return correctPasswords;
+    }
+
+    private static List<String> createWrongPasswords() {
+        List<String> wrongPasswords = new ArrayList<>();
+        wrongPasswords.add("");
+        wrongPasswords.add("a");
+        wrongPasswords.add("1");
+        wrongPasswords.add("A");
+        wrongPasswords.add("a1");
+        wrongPasswords.add("aA");
+        wrongPasswords.add("A1");
+        wrongPasswords.add("aa1");
+        wrongPasswords.add("a11");
+        wrongPasswords.add("AA1");
+
+        wrongPasswords.add(" aA1");
+        wrongPasswords.add("aA1 ");
+        wrongPasswords.add(" aA1 ");
+        wrongPasswords.add("a A1");
+
+        wrongPasswords.add("a`1");
+        wrongPasswords.add("A`1");
+        wrongPasswords.add("1#A");
+        wrongPasswords.add("!\"#$%&'()*+,-./:;<=>?@[\\]^_`{|}~");
+
+        wrongPasswords.addAll(getStringsWithRussianOrUkrCharacter("a", "A1"));
+
+        return wrongPasswords;
+    }
+
+    private static List<String> createCorrectPhoneNumbers() {
+        List<String> correctPhoneNumbers = new ArrayList<>();
+        correctPhoneNumbers.add("+38-050-223-32-23");
+        correctPhoneNumbers.add("38-050-223-32-23");
+        correctPhoneNumbers.add("     38  -  050  -  223  -  32  -  23     ");
+        correctPhoneNumbers.add("     38-050  -223-32-23     ");
+        correctPhoneNumbers.add("     38-050-223-   32-23     ");
+
+        correctPhoneNumbers.add("+38.050.223.32.23");
+        correctPhoneNumbers.add("38.050.223.32.23");
+        correctPhoneNumbers.add("     38.050.223.32.23     ");
+
```

Comment:
Why do we need this empty line?

Reply:
Just for convenience at the time of writing. I forgot to delete it. Thanks.